tool/tctl/common/resource_command_test.go

```diff
@@ -564,6 +679,17 @@ spec:
   target:
     user: "bad@actor"
   message: "Come see me"`
+
+	integrationYAML = `kind: integration
+sub_kind: aws-oidc
+version: v1
+metadata:
+  name: myawsint
+spec:
+  subkind_spec:
```

Comment:
Hmm do we really have "subkind_spec" field in the resource or is it just some test artifact? Why can't we just do
```
spec:
  aws_oidc:
    role_arn: ...
  somethin_else:
    ...
```

None of other resources had subkind_spec e.g. check out ProvisionToken, it also has a bunch of fields for different integrations (Github, AWS, etc.) and they're not nested under anything.

Reply:
Integration resource uses the OneOf definition (it was suggested here https://github.com/gravitational/teleport/pull/23553#discussion_r1150789478 )
So, when we MarshalJSON it comes with the OneOf type as a key:
```json
{
    "kind": "integration",
    "sub_kind": "aws-oidc",
    "version": "v1",
    "metadata": {
        "name": "some-integration"
    },
    "spec": {
        "subkind_spec": {
            "aws_oidc": {
                "role_arn": "arn:aws:iam::123456789012:role/DevTeams"
            }
        }
    }
}
```

I can customize the Un/MarshalJSON to "hide" the `subkind_spec`, but I would rather not add more custom code

Let me know if you still think we should remove that extra field and I will remove it :+1:

Reply:
If this was a non-user-facing resource, it wouldn't have mattered much but because users will have to create it the extra `subkind_spec` field just adds extra noise IMO.

With this structure we'd still have `spec.subkind_spec.aws_oidc`, `spec.subkind_spec.gcp_xxx` and so on, I don't really see how it's better than just `spec.aws_oidc`, `spec.gcp_xxx`, etc.

I agree we shouldn't just stuff all fields for all integrations in the integration's `spec` intermixed (that was main @rosstimothy's concern as I understood), but having separate sub-fields for different integrations like we have for provision tokens for example would make it easy to see which parameters are for which integration without extra noise.

Or I think you can keep using `oneof` but apply it to the whole spec? Something like this should work:

```protobuf
message IntegrationV1 {
  ResourceHeader Header = 1 [...];
  oneof Spec {
    AWSOIDCSpecV1 AWSOIDC = 2 [...];
    // Other integrations will go here
  }
}
```

Have you tried this?

Reply:
> Or I think you can keep using oneof but apply it to the whole spec?

I was trying to leave room for common properties
Things like status.


I'll change the serializers, that should be enough to get a better operator experience :+1: